libvast/src/system/connect_to_node.cpp

```diff
@@ -73,9 +74,22 @@ connect_to_node(scoped_actor& self, const caf::settings& opts) {
     auto& mm = self->system().middleman();
     return mm.remote_actor(node_endpoint.host, node_endpoint.port.number());
   }();
-  if (result)
-    VAST_VERBOSE(self, "successfully connected to", node_endpoint.host, ':',
-                 to_string(node_endpoint.port));
+  if (!result)
+    return result;
+  VAST_VERBOSE(self, "successfully connected to", node_endpoint.host, ':',
+               to_string(node_endpoint.port));
+  self
+    ->request(*result, defaults::system::initial_request_timeout, atom::get_v,
+              atom::version_v)
+    .receive(
+      [&](std::string node_version) {
+        if (node_version != VAST_VERSION)
+          VAST_WARNING(self,
+                       "Version mismatch between client and node detected; "
+                       "client: " VAST_VERSION,
+                       "node:", node_version);
+      },
+      [&](caf::error error) { result = std::move(error); });
```

Comment:
Considering this whole block does nothing if the log level is set to error or quiet at compile time, we should conditionally compile it.

```suggestion
#ifdef VAST_LOG_LEVEL >= VAST_LOG_LEVEL_WARNING
  self
    ->request(*result, defaults::system::initial_request_timeout, atom::get_v,
              atom::version_v)
    .receive(
      [&](std::string node_version) {
        if (node_version != VAST_VERSION)
          VAST_WARNING(self,
                       "Version mismatch between client and node detected; "
                       "client: " VAST_VERSION,
                       "node:", node_version);
      },
      [&](caf::error error) { result = std::move(error); });
#endif
```